tests/ut/ir/test_graph_function.py

```
@@ -0,0 +1,194 @@
+
+    assert plain.func_type == ir.FunctionType.Opaque
+    assert "graph_key" not in dict(plain.attrs)
+
+
+# ---------------------------------------------------------------------------
+# Rejected spellings
+# ---------------------------------------------------------------------------
+
+
+def test_graph_conflicts_with_explicit_type():
+    with pytest.raises(ParserTypeError, match="already implies type=pl.FunctionType.Graph"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)